scripts/tuple_tree_generator/tuple_tree_generator/generators/python.py

```diff
@@ -79,6 +93,125 @@ def _python_type(cls, resolved_type: Definition):
+        elif isinstance(field.resolved_type, ReferenceDefinition):
+            hint_type = "Reference"
+            ctor = "parse"
+        elif real_type in [
+            *(s.name for s in self.schema.struct_definitions() if s.abstract),
+            *(self.string_types),
+        ]:
+            hint_type = real_type
+            ctor = "parse"
+        else:
+            hint_type = real_type
+            ctor = "class"
+
+        assert hint_type
+
+        is_sequence = isinstance(field.resolved_type, SequenceDefinition)
+        if isinstance(field.resolved_type, SequenceDefinition):
+            is_abstract = isinstance(field.resolved_type.element_type, UpcastableDefinition)
+        else:
+            is_abstract = isinstance(field.resolved_type, UpcastableDefinition)
+
+        return (
+            f'{{"type": {hint_type}, '
+            + (f'"possibleValues": {possible_values},' if possible_values is not None else "")
+            + f'"ctor": "{ctor}", '
+            + f'"optional": {self.to_bool(field.optional)}, '
+            + f'"isArray": {self.to_bool(is_sequence)}, '
+            + f'"isAbstract": {self.to_bool(is_abstract)}}}'
+        )
+
+    @staticmethod
+    def to_bool(_input: bool) -> str:
+        return "True" if _input else "False"
+
+    def get_mixins(self, struct_name: str) -> str:
+        for mixins_path in self.mixins_paths:
+            with open(Path(mixins_path), "r") as mixins:
+                mixins_parsed = ast.parse(mixins.read())
+                mixin_classes = [
+                    node.name for node in mixins_parsed.body if isinstance(node, ast.ClassDef)
+                ]
+
+                for mc in mixin_classes:
+                    if struct_name == mc.removesuffix("Mixin"):
+                        return mc
+
+                if "AllMixin" in mixin_classes:
+                    return "AllMixin"
+        raise RuntimeError("Couldn't find matching Mixin class")
+
+    def get_mixins_imports(self) -> List[str]:
+        imports = []
+        for mixins_path in self.mixins_paths:
+            with open(Path(mixins_path), "r") as mixins:
+                mixins_parsed = ast.parse(mixins.read())
+                module_classes = [
+                    node.name for node in mixins_parsed.body if isinstance(node, ast.ClassDef)
+                ]
+                imports.append(
+                    f"from {self._get_mixins_import_path(mixins_path)} import {', '.join(module_classes)}"  # noqa: E501
+                )
+        return imports
+
+    @staticmethod
+    def _get_mixins_import_path(mixins_path: str) -> str:
+        # split the mixin path on the last occurrence of `revng` to
+        # get the root folder from where we import. Then remove the
+        # suffix and replace to make a proper import string.
+        revng_relative_path = mixins_path[mixins_path.rindex("revng"):]
+        return revng_relative_path.removesuffix(".py").replace("/", ".")
```

Comment:
This is too ad-hoc. There's a more robust way, by virtue of checking the output path of the template and figuring out the relative import of the mixin. What I mean is that you have:
```
PYTHON_PATH=${CMAKE_BINARY_DIR}/lib/python3.12/site-packages/revng/model/v1/_generated.py
MIXIN_PATH=${CMAKE_BINARY_DIR}/lib/python3.12/site-packages/revng/model/mixins.py
```
If you do `os.path.relpath(MIXIN_PATH, PYTHON_PATH)` you get `../../mixins.py` and you can derive the `from ..mixins` from that

Reply:
Do you mean `OUTPUT_PATH` instead of `PYTHON_PATH`? If yes, that arg is not supplied to the `python.py` script, it is an passed to [tuple_tree_generator_python.py](https://github.com/revng/revng/blob/develop/scripts/tuple_tree_generator/tuple-tree-generate-python.py#L44), should I pass the `output` arg to `python.py` too?

Reply:
> Do you mean OUTPUT_PATH instead of PYTHON_PATH?

Yes, my bad.

> should I pass the output arg to python.py too?

Yes, please do

Reply:
Let me know if the new approach is better.